pkg/api/middlewares_test.go

```diff
@@ -0,0 +1,71 @@
+package api
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	util_log "github.com/cortexproject/cortex/pkg/util/log"
+)
+
+var HTTPTestMiddleware = HTTPHeaderMiddleware{TargetHeaders: []string{"TestHeader1", "TestHeader2", "Test3"}}
+
+func TestHeaderInjection(t *testing.T) {
+	ctx := context.Background()
+	h := http.Header{}
+	contentsMap := make(map[string]string)
+	contentsMap["TestHeader1"] = "RequestID"
+	contentsMap["TestHeader2"] = "ContentsOfTestHeader2"
+	contentsMap["Test3"] = "SomeInformation"
+
+	h.Add("TestHeader1", contentsMap["TestHeader1"])
+	h.Add("TestHeader2", contentsMap["TestHeader2"])
+	h.Add("Test3", contentsMap["Test3"])
+
+	req := &http.Request{
+		Method:     "GET",
+		RequestURI: "/HTTPHeaderTest",
+		Body:       http.NoBody,
+		Header:     h,
+	}
+
+	req = req.WithContext(ctx)
+	ctx = HTTPTestMiddleware.InjectTargetHeadersIntoHTTPRequest(req)
+
+	headerMap, ok := ctx.Value(util_log.HeaderMapContextKey).(map[string]string)
+	require.True(t, ok)
+
+	for _, header := range HTTPTestMiddleware.TargetHeaders {
```

Comment:
how about the case where a header is in `contentsMap` but not in `TargetHeaders` ?